cas-client-core/src/main/java/org/jasig/cas/client/validation/Cas30ServiceTicketValidator.java

```diff
@@ -34,4 +42,31 @@ public Cas30ServiceTicketValidator(final String casServerUrlPrefix) {
     protected String getUrlSuffix() {
         return "p3/serviceValidate";
     }
+
+    /**
+     * Custom attribute extractor that will account for inlined CAS attributes.  Useful when CAS is acting as
+     * as SAML 2 IdP and returns SAML attributes with names that contains namespaces.
+     *
+     * @param xml the XML to parse.
+     * @return - Map of attributes
+     */
+    @Override
+    protected Map<String, Object> extractCustomAttributes(String xml) {
+        final Document document = XmlUtils.newDocument(xml);
+        final HashMap<String, Object> attributes = new HashMap<String, Object>();
+
+        NodeList attributeList = document.getElementsByTagName("cas:attribute");
+
+        for (int i = 0; i < attributeList.getLength(); i++) {
+            final Node attribute = attributeList.item(i);
+            if (attribute.getAttributes().getLength() > 0) {
```

Comment:
You may want to extract `attribute.getAttributes()` into something, so as to not repeat the call a few more times.